crates/sui/src/keytool.rs

```diff
@@ -254,210 +264,330 @@ pub enum KeyToolCommand {
+                    let (sui_address, kp) = get_authority_key_pair();
+                    let public_base64_key = kp.encode_base64();
+                    let file_name = format!("bls-{sui_address}.key");
+                    write_authority_keypair_to_file(&kp, file_name)?;
+                    CommandOutput::Generate(Key {
+                        sui_address,
+                        public_base64_key,
+                        key_scheme: key_scheme.to_string(),
+                        flag: SignatureScheme::BLS12381.flag(),
+                        mnemonic: None,
+                        peer_id: None,
+                    })
+                }
+                _ => {
+                    let (sui_address, kp, scheme, phrase) =
                         generate_new_key(key_scheme, derivation_path, word_length)?;
-                    let file = format!("{address}.key");
-                    write_keypair_to_file(&kp, &file)?;
-                    println!(
-                        "Created new keypair for address wrote to file path {:?} with scheme {:?}: [{address}]",
-                        file, scheme
-                    );
-                    println!("Secret Recovery Phrase : [{phrase}]");
+                    let public_base64_key = kp.encode_base64();
+                    let file = format!("{sui_address}.key");
+                    let peer_id = if let PublicKey::Ed25519(public_key) = kp.public() {
+                        Some(anemo::PeerId(public_key.0).to_string())
+                    } else {
+                        None
+                    };
+                    write_keypair_to_file(&kp, file)?;
+                    CommandOutput::Generate(Key {
+                        sui_address,
+                        public_base64_key,
+                        key_scheme: scheme.to_string(),
+                        flag: kp.public().flag(),
+                        mnemonic: Some(phrase),
+                        peer_id,
+                    })
                 }
-            }
+            },
             KeyToolCommand::Show { file } => {
                 let res = read_keypair_from_file(&file);
                 match res {
                     Ok(keypair) => {
-                        println!("Public Key: {}", keypair.public().encode_base64());
-                        println!("Flag: {}", keypair.public().flag());
-                        if let PublicKey::Ed25519(public_key) = keypair.public() {
-                            let peer_id = anemo::PeerId(public_key.0);
-                            println!("PeerId: {}", peer_id);
-                        }
+                        let sui_address: SuiAddress = (&keypair.public()).into();
+                        let peer_id = if let PublicKey::Ed25519(public_key) = keypair.public() {
+                            Some(anemo::PeerId(public_key.0).to_string())
+                        } else {
+                            None
+                        };
```

Comment:
is this part just to parse from a PublicKey::Ed25519 to the anemo style format display? seeing this a few times perhaps worth extracting to a common method.

Reply:
Yes, will do that!